providers/databricks/src/airflow/providers/databricks/hooks/databricks_base.py

```diff
@@ -264,11 +283,12 @@ async def _a_get_sp_token(self, resource: str) -> str:
 
         self.log.info("Existing Service Principal token is expired, or going to expire soon. Refreshing...")
         try:
+            conn = await self.a_databricks_conn()
```

Comment:
Style nit: this is more normally how we and python projects differenetiate a sync fn from an async one
```suggestion
            conn = await self.adatabricks_conn()
```